contracts/governance/FeeSharingCollector/FeeSharingCollector.sol

```diff
@@ -470,56 +520,110 @@ contract FeeSharingCollector is
     /**
      * @dev withdraw all of the RBTC balance based on particular checkpoints
      *
-     * RBTC balance consists of:
-     * - rbtc balance
-     * - wrbtc balance which will be unwrapped to rbtc
-     * - iwrbtc balance which will be unwrapped to rbtc
+     * This function will withdraw RBTC balance which is passed as _token param, so it could be either of these:
+     * - rbtc balance or
+     * - wrbtc balance which will be unwrapped to rbtc or
+     * - iwrbtc balance which will be unwrapped to rbtc or
      *
+     * @dev WARNING! This function skips all the checkpoints before '_fromCheckpoint' irreversibly, use with care
+     *
+     * @param _tokens array of either RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT or wrbtc address or iwrbtc address
      * @param _maxCheckpoints  Maximum number of checkpoints to be processed to workaround block gas limit
      * @param _receiver An optional tokens receiver (msg.sender used if 0)
      */
-    function withdrawRBTC(uint32 _maxCheckpoints, address _receiver) external nonReentrant {
-        _withdrawRBTC(_maxCheckpoints, _receiver);
+    function withdrawRbtcTokens(
```

Comment:
this function should accumulate RBTC and RBTC-base tokens fees and then transfer total RBTC amount to the user in one call

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/507/commits/d88f44fa441ae08c4fc117f5a169596574d95911)